[CS2113T-F08-3] Connoisseur

huachen24 opened the pull request:
Connoisseur helps people have an organized way to keep track of their experiences. It is optimized for CLI users so that frequent tasks can be done faster by typing in commands.

huachen24 committed:
fix checkstyle

huachen24 committed:
update deleteReview test

Krithigha24 committed:
Merge pull request #62 from huachen24/tidy-up

Convert txt storage to json storage

huachen24 committed:
add does not exist message to commands in reco mode

huachen24 committed:
add price range to recommendations

huachen24 committed:
update gitignore

huachen24 committed:
update git ignore

Krithigha24 committed:
Merge pull request #65 from huachen24/reco

Add price range to recommendation

Krithigha24 committed:
Added location recommendation
When viewing recommendation you can now see location and recby
increased spacing for title of review and recommendation

Krithigha24 committed:
checkstyle

huachen24 committed:
fix bug where preferred sort is not persistent

huachen24 committed:
Merge pull request #66 from Krithigha24/Krithigha-changesToRecoMode

Added location recommendation

huachen24 committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113T-F08-3/tp into master

Krithigha24 committed:
added border when listing recommendation and review and when viewing a title in review

Krithigha24 committed:
stylecheck

angjiahui committed:
Tidy up CommandList class.
Split commands in CommandList into RecommendationsCommandList and ReviewCommandList.
Move determineEditCommand into ReviewCommandList.

angjiahui committed:
Remove extra lines

huachen24 committed:
Merge pull request #67 from huachen24/master

fix bug where preferred sort is not persistent

EmilyTJX committed:
Merge pull request #68 from Krithigha24/Krithigha-changesToRecoMode

added border when listing recommendation and review and when viewing …

huachen24 committed:
Merge branch 'master' into JiaHui-SplitCommandList

huachen24 committed:
Merge pull request #69 from jhjhajh/JiaHui-SplitCommandList

Tidy up CommandList class.

huachen24 committed:
fix display whitespace issue

Krithigha24 committed:
Merge pull request #70 from huachen24/reco

fix display whitespace issue

huachen24 committed:
update ug

angjiahui committed:
Add images for UG

huachen24 committed:
update ug

huachen24 committed:
Merge pull request #71 from jhjhajh/JiaHui-UG

Add images for UG

huachen24 committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113T-F08-3/tp into ug

huachen24 committed:
add in pictures'

huachen24 committed:
Merge pull request #72 from huachen24/ug

Ug

huachen24 committed:
fix links

huachen24 committed:
Merge pull request #73 from huachen24/ug

fix links

Krithigha24 committed:
user guide changes

EmilyTJX committed:
Added in convertRecommendation to review function

huachen24 committed:
bug fixes

huachen24 committed:
Merge pull request #74 from Krithigha24/master

user guide changes

huachen24 committed:
Merge pull request #75 from huachen24/ug

bug fixes

EmilyTJX committed:
Update in convertRecommendation to review function

EmilyTJX committed:
Update in convertRecommendation to review function

EmilyTJX committed:
Added in Missing RecommendationList.java

EmilyTJX committed:
Checkstyle errors

EmilyTJX committed:
Checkstyle errors for recoList

EmilyTJX committed:
broken converter

EmilyTJX committed:
checkstyle broken converter

EmilyTJX committed:
update converter

angjiahui committed:
Bug Fixes and Standardize input
Standardize input format for recomendation and review
Fix blank inputs

angjiahui committed:
Fix style

angjiahui committed:
Fix style checks

huachen24 committed:
Merge pull request #77 from jhjhajh/JiaHui-BugFix

Standardize input for review and recommendations

huachen24 committed:
Merge branch 'master' into master

huachen24 committed:
Merge pull request #76 from EmilyTJX/master

Updated names and convertfucntion

Krithigha24 committed:
added edit feature to recommendation list

Krithigha24 committed:
checkstyle

angjiahui committed:
Update help commands for V2.0

huachen24 committed:
Merge pull request #79 from jhjhajh/JiaHui-HelpCommandsv2

Update help commands for V2.0

jalvinchan commented on docs/DeveloperGuide.md:
```diff
@@ -1,8 +1,46 @@
 ## Design & implementation
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+### Sorting
```
I think that it can have a class/sequence diagram here.

jalvinchan commented on docs/DeveloperGuide.md:
```diff
@@ -1,8 +1,46 @@
+    1. Follow the guide [[se-edu/guides] IDEA: Importing a Gradle project](https://se-education.org/guides/tutorials/intellijImportGradleProject.html) to import the project into IDEA
+
+3. **Verify the setup:**
+    1. Run the `seedu.address.connoisseur` and try a few commands.
```
Maybe suggest a few commands that can be run?

jalvinchan commented on docs/DeveloperGuide.md:
```diff
@@ -1,8 +1,46 @@
+### Purpose and Scope
+The purpose of this developer guide is to describe the architecture and software design decisions for our application. This guide will cover our program architecture, logical view of major components and functionalities of our features.
+
+## Setting up, getting started
```
Very detailed guide, good job!

jalvinchan commented on docs/DeveloperGuide.md:
```diff
@@ -1,8 +1,46 @@
+### Sorting
+1. The sorting component is used to sort the reviews using a variety of sorting methods. 
+2. A predefined number of sorting methods are enumerated and custom comparator methods are generated for them. 
+3. When a sorting method is chosen, the reviews will be sorted based on that method and returned to the UI be displayed. 
```
Maybe provide some examples?

Krithigha24 committed:
used existing ui instance

huachen24 committed:
Merge pull request #78 from Krithigha24/master

added edit feature to recommendation list

Krithigha24 committed:
user guide updates

Krithigha24 committed:
user guide updates

Krithigha24 committed:
user guide updates

Krithigha24 committed:
user guide updates

Krithigha24 committed:
User guide updates

Krithigha24 committed:
user guide final updates

huachen24 committed:
Merge pull request #80 from Krithigha24/Krithigha-UserGuideEdits

user guide updates

huachen24 committed:
tidy code and implement unlimited tries for invalid input

sarzorwyn commented on docs/DeveloperGuide.md:
```diff
@@ -1,8 +1,46 @@
+## Introduction
+
+### What is Connoisseur
+Connoisseur is a desktop app for managing and storing personal reviews that students would like to keep. It is a revolutionary app that not only provides an organised user friendly database for its users, but provides customisable categories to enhance the user experience. Through its intuitive command line interface, students would be able to store and share their recommendations easily.
```
Good introduction for your product

sarzorwyn commented on docs/DeveloperGuide.md:
```diff
@@ -1,8 +1,46 @@
+<div class="alert alert-block alert-info">
+<b>Note:</b>  Importing a Gradle project is slightly different from importing a normal Java project.</div>
+
 ## Design & implementation
```
I think you should prioritize expanding this part so that the readers of the dg will understand the flow of your program and the design choices

huachen24 committed:
Merge pull request #81 from huachen24/master

tidy code and implement unlimited tries for invalid input

huachen24 committed:
fix sortmethod not saving

angjiahui committed:
Restrict number of characters for some imput fields
Review: title, category
Recommendation: title, category, location, recommended by

Krithigha24 committed:
developer guide edits

Krithigha24 committed:
update

huachen24 committed:
Merge pull request #84 from jhjhajh/JiaHui-Bug-FixV2

Restrict number of characters for some input fields

EmilyTJX committed:
Fix Figure naming, links.
Add in command summary in UG

huachen24 committed:
Merge pull request #83 from huachen24/master

fix sortmethod not saving

Krithigha24 committed:
update

huachen24 committed:
Merge pull request #86 from EmilyTJX/master

Fix Figure naming, links.

huachen24 committed:
Merge pull request #85 from Krithigha24/Krithigha-DeveloperGuideEdits

developer guide edits

angjiahui committed:
Edit hyperlinks for UG

angjiahui committed:
no message

huachen24 committed:
Merge pull request #88 from jhjhajh/JiaHui-UGV2

Edit hyperlinks for UG

Krithigha24 committed:
UG edits

Krithigha24 committed:
Merge pull request #103 from Krithigha24/Krithigha-UserGuideEdits

UG edits

huachen24 committed:
bug fixes

huachen24 committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113T-F08-3/tp into master

huachen24 committed:
update

huachen24 committed:
add display stars command

huachen24 committed:
update javadocs

Krithigha24 committed:
Merge pull request #121 from huachen24/master

bug fixes

Krithigha24 committed:
bug fixes

Krithigha24 committed:
ug edits

Krithigha24 committed:
edits

Krithigha24 committed:
Merge pull request #124 from Krithigha24/Krithigha-UserGuideEdits

ug edits

Krithigha24 committed:
ug edits

EmilyTJX committed:
Merge pull request #125 from Krithigha24/Krithigha-UserGuideEdits

ug edits

EmilyTJX committed:
fixed hyperlinks in ug

Krithigha24 committed:
Merge pull request #127 from EmilyTJX/master

fixed hyperlinks in ug

Krithigha24 committed:
made edits

Krithigha24 committed:
Merge pull request #123 from Krithigha24/Krithigha-bugFixes

bug fixes

EmilyTJX committed:
fixed price double, loop and grid issue

EmilyTJX committed:
fixed price double, loop and grid issue

EmilyTJX committed:
checkstyle in recolist

huachen24 committed:
update naming

huachen24 committed:
add diagrams

angjiahui committed:
Add JSONException if file is corrupted

angjiahui committed:
Make view command case insensitive

angjiahui committed:
Restrict price range to whole numbers between 0-9999

angjiahui committed:
Make extra parameters invalid for mode switch and list in reco mode

angjiahui committed:
Add prompt when user tries to edit a non-existent recommendation

angjiahui committed:
Fix typo in UG

angjiahui committed:
Fix stylechecks

angjiahui committed:
Link github profile in about me

angjiahui committed:
-

angjiahui committed:
Fix bug for extra parameters and less than required parameters.

angjiahui committed:
Remove extra comments

huachen24 committed:
Merge pull request #129 from EmilyTJX/master

fixed price double, loop and grid issue

Krithigha24 committed:
Merge pull request #130 from huachen24/master

Add UML Diagrams

angjiahui committed:
Minor changes to README.md

huachen24 committed:
Merge branch 'master' into JiaHui-BugFixV2.1

huachen24 committed:
Update Ui.java

huachen24 committed:
Update Messages.java

huachen24 committed:
Update Messages.java

huachen24 committed:
Merge pull request #132 from jhjhajh/JiaHui-UGV2.1

Minor Changes to docs/

huachen24 committed:
Merge pull request #131 from jhjhajh/JiaHui-BugFixV2.1

Jia hui Bug Fix v2.1

EmilyTJX committed:
edit headings of dg

EmilyTJX committed:
add back user story

angjiahui committed:
Fix bugs from previous bugfix: Make edit, delete and done case insensitive

huachen24 committed:
Merge pull request #133 from EmilyTJX/master

edit headings of dg

angjiahui committed:
Standarize implementation

angjiahui committed:
Standardize to equalsIgnoreCase

huachen24 committed:
Merge pull request #134 from jhjhajh/JiaHui-BugFixV2.1

Fix bugs from previous bugfix:

Krithigha24 committed:
- price is now displayed with 2 decimal points and changed it to string type
- bug fixed : when adding a review/reco a duplicate title could be added if i typed "avengers " with a black space next to it

Krithigha24 committed:
checkstyle

huachen24 committed:
Merge pull request #135 from Krithigha24/master

slight bug fixes

huachen24 committed:
final edits

huachen24 committed:
fix typo

huachen24 committed:
make display a global command

huachen24 committed:
Merge pull request #136 from huachen24/master

Final fixes

angjiahui committed:
Fix link to sections in DeveloperGuide and Correct typo in Messages.java

huachen24 committed:
Merge pull request #137 from jhjhajh/JiaHui-DGV2.1

Fix link to sections in DeveloperGuide and Correct typo in Messages.java

huachen24 committed:
add main design

huachen24 committed:
fix list bugs

angjiahui committed:
Edit DG. Remove unused Empty Input Exception
Developer Guide: Add Exception Handling and Manual testing for launch and shutdown

Krithigha24 committed:
user guid edits

Krithigha24 committed:
Merge pull request #140 from Krithigha24/master

user guid edits

huachen24 committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113T-F08-3/tp into master

huachen24 committed:
update AboutUs

huachen24 committed:
Merge pull request #138 from huachen24/bugs

Fix latest bugs

huachen24 committed:
Merge pull request #141 from huachen24/master

update aboutus

Krithigha24 committed:
User Guide edits

Krithigha24 committed:
edits

Krithigha24 committed:
edits

Krithigha24 committed:
Merge pull request #142 from Krithigha24/master

User Guide edits

Krithigha24 committed:
bug fixes

Krithigha24 committed:
bug fixes that reflect in the user guide

Krithigha24 committed:
checkstyle

Krithigha24 committed:
checkstyle

EmilyTJX committed:
edit ugdg

EmilyTJX committed:
edit ugdg

huachen24 committed:
Merge pull request #143 from Krithigha24/Krithigha-bugFixes

bug fixes

huachen24 committed:
update design section of dg

huachen24 committed:
standardize diagram colours

huachen24 committed:
Merge pull request #144 from huachen24/master

update design section of dg

EmilyTJX committed:
edit ugdg

EmilyTJX committed:
edit ugdg

EmilyTJX committed:
add back old dg edits

huachen24 committed:
Merge pull request #145 from EmilyTJX/master

re-added old dg

EmilyTJX committed:
add reco dg

EmilyTJX committed:
list reco dg

EmilyTJX committed:
edit reco dg

angjiahui committed:
Add more specific headings and files for 'documentation, testing and dev-ops'

EmilyTJX committed:
review reco dg

EmilyTJX committed:
review reco dg

EmilyTJX committed:
add alignement ss in ug

huachen24 committed:
Merge pull request #146 from EmilyTJX/master

recommendation dg rough work

angjiahui committed:
Add Appendix A, B, C, E and F

angjiahui committed:
Remove printstacktrace()

Krithigha24 committed:
Developer Guide edits to section 5 to 5.2.6

huachen24 committed:
Merge pull request #147 from Krithigha24/Krithigha-DeveloperGuideEdits

Developer Guide edits to section 5 to 5.2.6

huachen24 committed:
add storage part

EmilyTJX committed:
draft for PPP

EmilyTJX committed:
draft for PPP

EmilyTJX committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113T-F08-3/tp

huachen24 committed:
Merge pull request #149 from EmilyTJX/master

draft for PPP

angjiahui committed:
Add more manual testing cases in DG

angjiahui committed:
Updated Appendix D, manual testing

angjiahui committed:
Edit PPP and minor changes to commandstest

angjiahui committed:
Minor edits to PPP

angjiahui committed:
Add tests for ReviewList

angjiahui committed:
Add tests for recommendationList

angjiahui committed:
Fix stylechecks

huachen24 committed:
Merge pull request #148 from huachen24/master

add storage part

huachen24 committed:
Merge pull request #151 from jhjhajh/JiaHui-test

Added some jUnit tests

huachen24 committed:
Merge branch 'master' into JiaHui-PPP

huachen24 committed:
Merge pull request #150 from jhjhajh/JiaHui-PPP

Edit PPP and minor changes to commandstest

huachen24 committed:
Merge branch 'master' into JiaHui-DGV2.1

huachen24 committed:
Update DeveloperGuide.md

huachen24 committed:
fix style

huachen24 committed:
Merge pull request #139 from jhjhajh/JiaHui-DGV2.1

Edit DG. Remove unused Empty Input Exception

huachen24 committed:
remove unused imports

Krithigha24 committed:
PPP

huachen24 committed:
update PPP

huachen24 committed:
update PPP

huachen24 committed:
Merge pull request #152 from Krithigha24/Krithigha-DeveloperGuideEdits

draft PPP - Krithigha

Krithigha24 committed:
DG edits Pending : need to fix 2 anchor issues and update how to use this guide

Krithigha24 committed:
Merge pull request #154 from Krithigha24/Krithigha-DeveloperGuideEdits

DG edits Pending : need to fix 2 anchor issues and update how to use …

angjiahui committed:
add github username

Krithigha24 committed:
edits

huachen24 committed:
UG updates

angjiahui committed:
fix stylechecks

Krithigha24 committed:
Merge pull request #156 from Krithigha24/Krithigha-DeveloperGuideEdits

DG edits

EmilyTJX committed:
updates to UG

EmilyTJX committed:
updates to UG

huachen24 committed:
Merge pull request #153 from huachen24/master

PPP

angjiahui committed:
Code authorship in commands test and remove unused imports

Krithigha24 committed:
Added author tags

EmilyTJX committed:
ug edits

angjiahui committed:
Update PPP and Links to AboutUs

Krithigha24 committed:
checkstyle

EmilyTJX committed:
Merge branch 'master' into master

jhjhajh committed:
Merge pull request #157 from Krithigha24/Krithigha-DeveloperGuideEdits

Added author tags

Krithigha24 committed:
Merge pull request #159 from jhjhajh/JiaHui-PPP

Update PPP and Links to AboutUs

jhjhajh committed:
Merge pull request #155 from jhjhajh/JiaHui-codeAuthorship

add github username

angjiahui committed:
Edit links in AboutUs

huachen24 committed:
Merge pull request #160 from jhjhajh/JiaHui-PPP

Update links in AboutUs

Krithigha24 committed:
about us update

EmilyTJX committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113T-F08-3/tp

huachen24 committed:
Merge pull request #158 from EmilyTJX/master

UG edits

huachen24 committed:
Merge pull request #161 from Krithigha24/Krithigha-AboutUs

about us update

huachen24 committed:
revert dg

huachen24 committed:
Merge pull request #162 from huachen24/master

revert dg

EmilyTJX committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113T-F08-3/tp

angjiahui committed:
minor edits to ug and fixed null input for command 'view'

angjiahui committed:
Fix sentence structure

huachen24 committed:
minor dg edits

Krithigha24 committed:
included some trims - minor bugs

Krithigha24 committed:
edits to my ppp

huachen24 committed:
update readme and profile photo

EmilyTJX committed:
dg edits

huachen24 committed:
change path to photos

Krithigha24 committed:
Merge pull request #165 from Krithigha24/Krithigha-bugFixes

included some trims - minor bugs

EmilyTJX committed:
final dg edits

huachen24 committed:
Merge pull request #163 from jhjhajh/Final-Changes

Final changes for UG and Fixed 'view' bug

huachen24 committed:
Merge pull request #164 from huachen24/master

minor dg edits

huachen24 committed:
Merge branch 'master' into master

huachen24 committed:
Merge pull request #166 from EmilyTJX/master

DG edits

EmilyTJX committed:
edit PPP

huachen24 committed:
regenerate diagrams

huachen24 committed:
Merge pull request #168 from huachen24/master

regenerate diagrams

EmilyTJX committed:
Merge pull request #167 from EmilyTJX/master

edit PPP

huachen24 committed:
#

huachen24 committed:
update links
